llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -108,6 +109,9 @@ bool VPRecipeBase::mayReadFromMemory() const {
   case VPWidenLoadEVLSC:
   case VPWidenLoadSC:
     return true;
+  case VPInterleaveEVLSC:
+  case VPInterleaveSC:
+    return cast<VPInterleaveBase>(this)->getNumStoreOperands() == 0;
```

Comment:
Are we missing mayReadFromMemory for regular VPInterleaveRecipes? Can this be split off into a separate PR?